File: uke0/baathus-109/Baathus.java
class Baathus {

    private int nPlasser;
    private int nBaater = 0;
    private Baat[] baatList;
    
    public Baathus(int antallPlasser) {
	baatList = new Baat[antallPlasser];
	nPlasser = antallPlasser;
    }

    public void settInn(String merke) {
	if (nBaater >= nPlasser) {
	    System.out.println("Kan ikke sette inn ny båt, båthuset er fullt.");
	}
	else {
	    baatList[nBaater] = new Baat(merke);
	    nBaater++;
	}
    }
    
    public void skrivBaater() {
	for (int i = 0; i < nBaater; i++) {
	    System.out.println(baatList[i].hentInfo());
	}
    }
}

File: uke0/baathus-109/c++/BaatHus.cpp
#include "BaatHus.h"
#include <iostream>

BaatHus::BaatHus(int kapasitet) : m_kapasitet(kapasitet), m_nBaater(0)
{
  std::cout << "Baathus created!" << std::endl;
}
BaatHus::~BaatHus()
{  
  std::cout << "Baathus destructed!" << std::endl;
}
void BaatHus::settInn(const char* navn)
{
  if (m_nBaater < m_kapasitet)
    {
      baater.push_back(Baat(navn));
      m_nBaater++;
    }
  else
    {
      std::cout << "Det er fullt!" << std::endl;
    }
}
void BaatHus::skrivBaater()
{
  for (Baat b : baater)
    {
      std::cout << b.hentInfo() << std::endl;
    }
}
